Coin market cap

githubRaj opened the pull request:
Created the CoinMarketCap Repo as backup, as requested through issue #66 . Testing done on whether or not data is retrieved successfully as well, in accordance with the maintainability of the app (issue #39 ).

githubRaj committed:
CoinMarketCap API Repo

Created the CoinMarketCap Repo, as requested through issue #66.

githubRaj committed:
Testing if API data is retrieved successfully

Test added for API data retrieval.

githubRaj committed:
Get All Data Test

Testing if all data is obtained when no stock ID is provided. Issues: #66 , #39.

GurkomalSRao commented on src/main/java/nitezh/ministock/dataaccess/CoinMarketCapRepo.java:
```diff
@@ -0,0 +1,37 @@
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
```
remove `import org.json.JSONObject;`
unnecessary import

githubRaj replied:
Done

GurkomalSRao commented on src/test/java/nitezh/ministock/dataaccess/CoinMarketCapRepoTest.java:
```diff
@@ -19,11 +20,24 @@
-    public void getAPIData() throws JSONException{
-        String id = "bitcoin";
+    public void getAllAPIData() throws JSONException{
+        String blankID = "";  //should get all the data if no id is provided
```
this only tests for top 100 cryptocurrencies. test whether you can retrieve all 1500 values

Add another test to validate each individual variable within the JSON Object/Array. You may retrieve bitcoin but the prices,marketcap or volume may be empty.

githubRaj committed:
Requested changes

Made the requested Changes, modified exisitng tests related to CoinCapMarket and created a new test to test the limit feature.

GurkomalSRao reviewed (changes requested):
requested changes are in comments
